shared/src/main/scala/mlscript/NewParser.scala

```diff
@@ -297,10 +304,23 @@ abstract class NewParser(origin: Origin, tokens: Ls[Stroken -> Loc], raiseFun: D
       case (SPACE, _) :: _ => consume; block
       case c =>
         val t = c match {
+          case ModifierSet(mods, (KEYWORD("constructor"), l0) :: c) =>
```

Comment:
I don't understand why `constructor` is considered a modifier. It would only make sense if we wrote `constructor fun` or other modified declaration...